### ISSUE TITLE
[examples/quadrotor] QuadrotorPlant dynamics are incorrect.

### ISSUE THREAD

**Issue by RussTedrake** [RobotLocomotion/drake#17246]
The unit test in `quadrotor_dynamics_test.cc` has the line:
```
  EXPECT_TRUE(ge_pose.rotation().IsNearlyEqualTo(ge_pose.rotation(), 1e-10));
```
Unsurprisingly, this passes.  But if you change it to the intended
```
  EXPECT_TRUE(ge_pose.rotation().IsNearlyEqualTo(mb_pose.rotation(), 1e-10));
```
Then the test fails.

I found this because I was writing up a (not so) quick example of the 3D dynamics of a quadrotor using Propeller.  Which gives different answers.  In particular, the dynamics don't match if I have non-zero angular velocity.

The multibody notation in [this line](https://github.com/RobotLocomotion/drake/blob/53571e1d65d1716d8b4b72ae8d3571ee89fa36cf/examples/quadrotor/quadrotor_plant.cc#L107) is wrong:
```
const Vector3<T> w_BN_B = rpy.CalcAngularVelocityInChildFromRpyDt(rpyDt);
```
According to that method's documentation, it should return `w_NB_B`, not `w_BN_B`.  Unfortunately, I'm timing out now.  Can I ask for the dynamics team to resolve it the existing failing test in `quadrotor_dynamics_test.cc`, please?

cc @sherm1 in case a different assignee is more appropriate.  
cc @mpetersen94 , in case it has any bearing on the strange results you've been seeing.

**Comment by sherm1** [comment]
Paul @mitiguy can you figure out what's going on here?

**Comment by mitiguy** [comment]
@sherm1   I am starting to look at it now.

**Comment by mitiguy** [comment]
Closed by PR #17254
